src/Nethermind/Nethermind.JsonRpc.Test/Modules/TraceRpcModuleTests.cs

```diff
@@ -0,0 +1,451 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using Nethermind.Blockchain.Comparers;
+using Nethermind.Blockchain.Processing;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Blockchain.Rewards;
+using Nethermind.Blockchain.Tracing;
+using Nethermind.Blockchain.Validators;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Specs;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Crypto;
+using Nethermind.Db;
+using Nethermind.Db.Blooms;
+using Nethermind.Int256;
+using Nethermind.Evm;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.JsonRpc.Modules.Trace;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.State.Repositories;
+using Nethermind.TxPool;
+using NUnit.Framework;
+using BlockTree = Nethermind.Blockchain.BlockTree;
+using Nethermind.Blockchain.Find;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Serialization.Json;
+using Nethermind.Trie.Pruning;
+
+namespace Nethermind.JsonRpc.Test.Modules
+{
+    [Parallelizable(ParallelScope.All)]
+    [TestFixture]
+    public class TraceRpcModuleTests
+    {
+        private class Context
+        {
+            private readonly IBlockTree _blockTree;
+            AutoResetEvent resetEvent = new(false);
+
+            public Context(bool auRa = false)
+            {
+                IDbProvider dbProvider = TestMemDbProvider.Init();
```

Comment:
Couldn't we use TestBlockchain here?

Reply:
Yeah, probably we can. I just used what I found in existing tests: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.JsonRpc.Test/Modules/TraceModuleTests.cs#L69

Let me see if we can refactor it to TestBlockchain quickly